docs/jdbc-mariadb.md

```diff
@@ -96,6 +96,59 @@ Note: a non-empty string value for the `password` property must be set. While th
 be ignored when connecting with the Cloud SQL Connector using IAM auth, leaving it empty will cause
 driver-level validations to fail.
 
+### Service Account Impersonation
+
+The Java Connector supports service account impersonation with the
+`targetPrincipal` JDBC connection property. When enabled,
+all API requests are made impersonating the supplied service account. The
+IAM principal must have the iam.serviceAccounts.getAccessToken permission or
+the role roles/iam.serviceAccounts.serviceAccountTokenCreator.
+
+Example:
+```java
+    // Set up URL parameters
+    String jdbcURL = String.format("jdbc:mariadb://ignoreme:123/%s", DB_NAME);
+    Properties connProps = new Properties();
+    connProps.setProperty("user", "mysql-iam-user@gmail.com");
+    connProps.setProperty("password", "password");
+    connProps.setProperty("sslmode", "disable");
+    connProps.setProperty("socketFactory", "com.google.cloud.sql.mariadb.SocketFactory");
+    connProps.setProperty("cloudSqlInstance", "project:region:instance");
+    connProps.setProperty("enableIamAuth", "true");
+    connProps.setProperty("targetPrincipal", "mysql-iam-user@gmail.com");
```

Comment:
cloudSqlTargetPrincipal? Same with cloudSqlDelegates.

Reply:
I would rather leave this as is, and then create a separate PR where I combine all the docs into two pages: a page for R2DBC  and a page for JDBC.